GetStats crashes when called on Closed agent

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x200 pc=0x13124b2]

goroutine 4 [running]:
github.com/pion/ice.(*Agent).ok(0x0, 0xc000058140, 0x1d40008)
        /Users/seaduboi/go/pkg/mod/github.com/pion/ice@v0.7.8/agent.go:164 +0x22
github.com/pion/ice.(*Agent).run(0x0, 0xc000058140, 0xc00026a120, 0x0)
        /Users/seaduboi/go/pkg/mod/github.com/pion/ice@v0.7.8/agent.go:725 +0x40
github.com/pion/ice.(*Agent).GetCandidatePairsStats(0x0, 0x0, 0x0, 0x0)
        /Users/seaduboi/go/pkg/mod/github.com/pion/ice@v0.7.8/agent.go:1190 +0x8c
github.com/pion/webrtc/v2.(*ICEGatherer).collectStats.func1(0xc0000d3f10, 0xc00000e0c0)
        /Users/seaduboi/go/src/github.com/pion/webrtc/icegatherer.go:289 +0x47
created by github.com/pion/webrtc/v2.(*ICEGatherer).collectStats
        /Users/seaduboi/go/src/github.com/pion/webrtc/icegatherer.go:288 +0xc1
exit status 2
FAIL    github.com/pion/webrtc/v2       1.229s
```

--- a/icegatherer.go
+++ b/icegatherer.go
@@ -277,10 +277,14 @@ func (g *ICEGatherer) SignalCandidates() error {
 }
 
 func (g *ICEGatherer) collectStats(collector *statsReportCollector) {
-	collector.Collecting()
+	agent := g.getAgent()
+	if agent == nil {
+		return
+	}
 
-	go func(collector *statsReportCollector) {
-		for _, candidatePairStats := range g.agent.GetCandidatePairsStats() {
+	collector.Collecting()
+	go func(collector *statsReportCollector, agent *ice.Agent) {
+		for _, candidatePairStats := range agent.GetCandidatePairsStats() {
 			collector.Collecting()
 
 			state, err := toStatsICECandidatePairState(candidatePairStats.State)
@@ -326,7 +330,7 @@ func (g *ICEGatherer) collectStats(collector *statsReportCollector) {
 			collector.Collect(stats.ID, stats)
 		}
 
-		for _, candidateStats := range g.agent.GetLocalCandidatesStats() {
+		for _, candidateStats := range agent.GetLocalCandidatesStats() {
 			collector.Collecting()
 
 			networkType, err := getNetworkType(candidateStats.NetworkType)
@@ -356,7 +360,7 @@ func (g *ICEGatherer) collectStats(collector *statsReportCollector) {
 			collector.Collect(stats.ID, stats)
 		}
 
-		for _, candidateStats := range g.agent.GetRemoteCandidatesStats() {
+		for _, candidateStats := range agent.GetRemoteCandidatesStats() {
 			collector.Collecting()
 			networkType, err := getNetworkType(candidateStats.NetworkType)
 			if err != nil {
@@ -384,5 +388,5 @@ func (g *ICEGatherer) collectStats(collector *statsReportCollector) {
 			collector.Collect(stats.ID, stats)
 		}
 		collector.Done()
-	}(collector)
+	}(collector, agent)
 }

--- a/stats_go_test.go
+++ b/stats_go_test.go
@@ -297,3 +297,12 @@ func TestPeerConnection_GetStats(t *testing.T) {
 	assert.NoError(t, offerPC.Close())
 	assert.NoError(t, answerPC.Close())
 }
+
+func TestPeerConnection_GetStats_Closed(t *testing.T) {
+	pc, err := NewPeerConnection(Configuration{})
+	assert.NoError(t, err)
+
+	assert.NoError(t, pc.Close())
+
+	pc.GetStats()
+}

